### PR TITLE
[Sprint #6] User Account Page & its test code

### PR DESCRIPTION
**Pull request by BOBTommy**
---- NOTE ----
1. User Account Page Feature is added
2. User Account View code is added
3. urls & simple template page for account is added
4. Test code for AccountView is added

### CODE REVIEW

File: buildbuild/buildbuild/urls.py [hunk 1]
```diff
@@ -16,6 +18,7 @@
     url(r'^$', Home.as_view(), name='home'),
     url(r'^login/', Login.as_view(), name="login"),
     url(r'^logout/', Logout.as_view(), name="logout"),
+    url(r'^account/', login_required(AccountView.as_view(),login_url='/login'), name="account"),
```
Comment (dobestan) [inline buildbuild/buildbuild/urls.py]: - why `login_url` has added?
- it is currently set on `settings.py` :

```
LOGIN_URL = "/login/"
```

File: buildbuild/users/templates/users/account.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html>
+<head lang="en">
+    <meta charset="UTF-8">
+    <title></title>
+</head>
+<body>
+{% if user.is_authenticated %}
```
Comment (dobestan) [inline buildbuild/users/templates/users/account.html]: - why `user.is_authenticated` is needed?
- we currently wrapped this view with `login_required` so always render this view when user is **already authenticated**.

File: buildbuild/users/templates/users/account.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
```
Comment (dobestan) [inline buildbuild/users/templates/users/account.html]: - and new html view is not needed any more.
- we are going to use `extends base.html`.
- example at `login.html`

File: buildbuild/buildbuild/urls.py (hunk 1 above)
Comment (BOBTommy) [inline buildbuild/buildbuild/urls.py]: Oh. I didn't know that. I fix it

File: buildbuild/users/views.py [hunk 4]
```diff
@@ -54,3 +57,13 @@ def get_redirect_url(self):
         logout(self.request)
         messages.add_message(self.request, messages.SUCCESS, "Successfully Logout")
         return reverse("home")
+
+class AccountView(DetailView):
+    model = User
+    template_name = 'users/account.html'
+
+    slug_field = "email"
```
Comment (dobestan) [inline buildbuild/users/views.py]: - `slug_field` is not needed. implementation on `urls.py` has no slug field ( just simple `/accounts` )

File: buildbuild/users/views.py [hunk 5]
```diff
@@ -54,3 +57,13 @@ def get_redirect_url(self):
...
+    context_object_name = "user_account"
```
Comment (dobestan) [inline buildbuild/users/views.py:91]: - how about use simple `user` than `user_account` ?

File: buildbuild/users/views.py [hunk 6]
```diff
@@ -54,3 +57,13 @@ def get_redirect_url(self):
...
+
+    def get_object(self, queryset=None):
+        return User.objects.get(email = self.request.session['email'])
```
Comment (dobestan) [inline buildbuild/users/views.py]: - why this `get_object` is needed?
- in this case, we find user twice ( duplication on implemenation )
- so how about to change like this : 

```
def get_object(self, queryset = None)
    return self.request.user
```
- or you can just remove this function ( `get_object` ) and change this View to inherit `TemplateView`

File: buildbuild/users/templates/users/account.html (hunk 3 above)
Comment (BOBTommy) [inline buildbuild/users/templates/users/account.html]: Hmm.. it is more fit. I fix it

File: buildbuild/users/views.py (hunk 6 above)
Comment (BOBTommy) [inline buildbuild/users/views.py]: In detailView, generic view requires overriding get_object method

Comment (BOBTommy) [inline buildbuild/users/views.py]: If we change to TemplateView, it will be possible but, I'm not sure

File: buildbuild/users/views.py (hunk 4 above)
Comment (BOBTommy) [inline buildbuild/users/views.py]: Oops

File: buildbuild/users/views.py (hunk 6 above)
Comment (dobestan) [inline buildbuild/users/views.py]: so, if we maintain `DetailView` ( i agree with your idea that this is more explicit way. ), how about to change `get_object` implementation.

```
def get_object(self):
    return self.request.user
```

in this case, we can access user via session directly.

**Pull request closed without merging**